pytest_html/plugin.py

```diff
@@ -236,6 +239,7 @@ def append_extra_html(self, extra, extra_index, test_index):
                     content = content.decode("utf-8")
                 if self.self_contained:
                     href = data_uri(content)
+                    download = extra.get("name")
```

Comment:
You should add extension to the filename. Browsers seem to figure it out based on mime type but it is probably best to be explicit.

Reply:
Done (I think)